### PR TITLE
[FIX] Update visitor info on email reception based on current inbox settings

### PR DESCRIPTION
**Pull request by KevLehman**

### CODE REVIEW

File: server/features/EmailInbox/EmailInbox.ts [hunk 1]
```diff
@@ -38,7 +40,7 @@ export async function configureEmailInboxes(): Promise<void> {
 			tlsOptions: {
 				rejectUnauthorized: false,
 			},
-			// debug: (...args: any[]): void => console.log(...args),
+			// debug: (...args: any[]): void => logger.debug(args),
```
Comment (murtaza98) [inline server/features/EmailInbox/EmailInbox.ts:43]:
```suggestion
```

Do we need this comment?

Comment (KevLehman) [reply server/features/EmailInbox/EmailInbox.ts:43]: I mean, it's useful for local development, but leave it there mainly to let others know it's possible haha

**Review by murtaza98 — APPROVED** [review]

File: server/features/EmailInbox/EmailInbox.ts [hunk 2]
```diff
@@ -20,14 +21,15 @@ export async function configureEmailInboxes(): Promise<void> {
...
+	logger.info('Clearing old email inbox registrations');
 	for (const { imap } of inboxes.values()) {
 		imap.stop();
 	}
 
 	inboxes.clear();
 
 	for await (const emailInboxRecord of emailInboxesCursor) {
-		console.log('Setting up email interceptor for', emailInboxRecord.email);
+		logger.info('Setting up email interceptor for', emailInboxRecord.email);
```
Comment (sampaiodiego) [inline server/features/EmailInbox/EmailInbox.ts:32]: this usage will be deprecated (we need to add a warning for that)

please log either an object or use placeholders: https://github.com/pinojs/pino/blob/master/docs/api.md#message-string

File: server/features/EmailInbox/EmailInbox.ts [hunk 3]
```diff
@@ -54,8 +56,9 @@ export async function configureEmailInboxes(): Promise<void> {
 			try {
 				await EmailMessageHistory.insertOne({ _id: email.messageId, email: emailInboxRecord.email });
 				onEmailReceived(email, emailInboxRecord.email, emailInboxRecord.department);
-			} catch (e) {
+			} catch (e: any) {
 				// In case the email message history has been received by other instance..
+				logger.error(e.message);
```
Comment (sampaiodiego) [inline server/features/EmailInbox/EmailInbox.ts:61]: if you have an error object, you can log it entirely
```
logger.error(e);
```

if for some reason TS complains about it, you can add the error object to the `err` key
```
logger.error({ err: e });
```

File: server/features/EmailInbox/EmailInbox_Incoming.ts [hunk 4]
```diff
@@ -205,6 +221,7 @@ export async function onEmailReceived(email: ParsedMail, inbox: string, departme
 			},
 		});
 	}).catch((error) => {
-		Livechat.logger.log('Error receiving Email: %s', error.message);
+		console.log(error);
```
Comment (sampaiodiego) [inline server/features/EmailInbox/EmailInbox_Incoming.ts:224]: please avoid using `console.log`

File: server/features/EmailInbox/EmailInbox_Incoming.ts [hunk 5]
```diff
@@ -205,6 +221,7 @@ export async function onEmailReceived(email: ParsedMail, inbox: string, departme
...
+		Livechat.logger.error('Error receiving Email: %s', error.message);
```
Comment (sampaiodiego) [inline server/features/EmailInbox/EmailInbox_Incoming.ts:225]: same here with logging the error object. if you need to add a message to the log, you can add the `msg` key:

```
Livechat.logger.error({
  msg: 'Error receiving Email',
  err: error,
});
```

File: server/features/EmailInbox/EmailInbox_Incoming.ts [hunk 6]
```diff
@@ -30,12 +31,20 @@ const language = settings.get('Language') || 'en';
 const t = (s: string): string => TAPi18n.__(s, { lng: language });
 
 function getGuestByEmail(email: string, name: string, department?: string): any {
+	logger.debug(`Attempt to register a guest for ${ email } on department: ${ department }`);
 	const guest = LivechatVisitors.findOneGuestByEmailAddress(email);
 
 	if (guest) {
+		logger.debug(`Guest with email ${ email } found with id ${ guest._id }`);
+		if (guest.department !== department) {
+			logger.debug(`Switching departments for guest ${ guest._id }. Previous: ${ guest.department } New: ${ department }`);
```
Comment (sampaiodiego) [inline server/features/EmailInbox/EmailInbox_Incoming.ts:40]: for debugging purposes might be useful to add extra data, for this it is useful logging big objects with an additional message, like:
```js
logger.debug({
  msg: 'Switching departments for guest',
  guest,
  new: department,
});
```

File: server/features/EmailInbox/EmailInbox_Incoming.ts [hunk 7]
```diff
@@ -111,10 +122,14 @@ export async function onEmailReceived(email: ParsedMail, inbox: string, departme
 
 	const thread = references?.[0] ?? email.messageId;
 
+	logger.debug(`Fetching guest for visitor ${ email.from.value[0].address }`);
 	const guest = getGuestByEmail(email.from.value[0].address, email.from.value[0].name, department);
 
-	let room = LivechatRooms.findOneByVisitorTokenAndEmailThread(guest.token, thread, {});
+	logger.debug(`Guest ${ guest._id } obtained. Attempting to find or create a room on department ${ department }`);
+	let room = LivechatRooms.findOneByVisitorTokenAndEmailThreadAndDepartment(guest.token, thread, department, {});
+	logger.debug(`Room ${ room?._id } found for guest ${ guest._id }`);
```
Comment (sampaiodiego) [inline server/features/EmailInbox/EmailInbox_Incoming.ts:130]: same here with debugging info:
```js
logger.debug({
  msg: 'Room found for guest',
  room,
  guest,
});

**Review by sampaiodiego — COMMENTED** [review]
Although it is already merged some changes would be good to do.

**Comment by KevLehman** [comment]
Hey @sampaiodiego , i wasn't aware of those new conventions for logs, would be good to create a "log documentation" page somewhere to describe conventions and best practices, wdyt? 👀

**Comment by sampaiodiego** [comment]
> Hey @sampaiodiego , i wasn't aware of those new conventions for logs, would be good to create a "log documentation" page somewhere to describe conventions and best practices, wdyt? 👀

definitely a good idea.. I'll try to draft something 👍

**Pull request merged**